src/test/java/com/codurance/lightaccess/PGOneToManyShould.java

```diff
@@ -0,0 +1,82 @@
+package com.codurance.lightaccess;
+
+import org.junit.Test;
+import java.util.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsEqual.equalTo;
+public class PGOneToManyShould {
+
+    private static final int SOME_KEY = 1;
+    private static final Integer SOME_OTHER_KEY = 2;
+    public static final int SOME_VALUE = 3;
+    private static final Integer SOME_OTHER_VALUE = 4;
+
+    @Test
+    public void collectNothingWhenNoPut() throws Exception {
+        PGOneToMany<Integer, Integer> pgOneToMany = new PGOneToMany<>();
+
+        assertThat(pgOneToMany.collect((key, values) -> new DummyOneToMany(key,values)),
+                equalTo(new ArrayList<>()));
+    }
+
+    @Test
+    public void collectOneToNothingAfterPutKeyWithNoValue() throws Exception {
+        PGOneToMany<Integer, Integer> pgOneToMany = new PGOneToMany<>();
+
+        pgOneToMany.put(new PGKeyValue<>(SOME_KEY, Optional.empty()));
+
+        assertThat(pgOneToMany.collect((key,values) -> new DummyOneToMany(key,values)),
+                equalTo(Arrays.asList(new DummyOneToMany(SOME_KEY, new ArrayList<>()))));
+    }
+
+    @Test
+    public void collectOneToManyAfterManyPutForSameKey() throws Exception {
+        PGOneToMany<Integer, Integer> pgOneToMany = new PGOneToMany<>();
+
+        pgOneToMany.put(new PGKeyValue<>(SOME_KEY, Optional.of(SOME_VALUE)));
+        pgOneToMany.put(new PGKeyValue<>(SOME_KEY, Optional.of(SOME_OTHER_VALUE)));
+
+        assertThat(pgOneToMany.collect((key,values) -> new DummyOneToMany(key,values)),
+                equalTo(Arrays.asList(new DummyOneToMany(SOME_KEY, Arrays.asList(SOME_VALUE, SOME_OTHER_VALUE)))));
+    }
+
+    @Test
+    public void collectMultipleOneToManyAfterPutForDifferentKeys() throws Exception {

```

Comment:
You can combine these this test we the above one and change the name to `collectMappedValuesAssosiatedWithGivenKeys`